CalculatorIdo/Script.js

```diff
@@ -0,0 +1,149 @@
+            currString = document.calcForm.input.value;
+        }
+    } else {
+        document.calcForm.input.value = currString.substring(0, currString.length - 1);
+    }
+}
+
+insert = (input) => { 
+    if(isResultShown) {
+        isResultShown = false;
+        clearCalculator();
+    }
+
+    document.calcForm.input.value = document.calcForm.input.value + input;
+}
+
+equal = () => {
+    if(isResultShown) {
+        isResultShown = false;
+        clearCalculator();
+    }
+
+    let resultString = document.calcForm.input.value;
+    if(resultString) {
+        let calculatedResult = calculateResult(stringResultToArray(resultString));
+        if(calculatedResult === undefined || isNaN(calculatedResult)) {
+            alert('Syntax error');
+        } else {
+            document.calcForm.input.value = calculatedResult;
+            currAns = calculatedResult;
+            isResultShown = true;
+            isDotAvailable = true;
+        }
+    }
+}
+
+stringResultToArray = (resultString) => {
+    let resultArray = [];
+    let currResult = '';
+    let currChar = '';
+
+    for (let i = 0; i < resultString.length; i++) {
+        currChar = resultString.charAt(i);
+        
+        if (/^[)(*/+-]$/.test(currChar)) {
+            if (currResult === '' && currChar === '-' &&
+                !(i > 0 && resultString.charAt(i-1) === ')')) {
+                currResult = '-';
+            } else {
+                if(currChar === '(' ||
+                   (/^[)*/+-]$/.test(currChar) && currResult === '')) {
+                    resultArray.push(currChar);
+                } else {
+                    resultArray.push(parseFloat(currResult), currChar);
+                }
+                currResult = '';
+            }
+        } else {
+            currResult += currChar;
+        }
+    }
+
+    if (currResult !== '') {
+        resultArray.push(parseFloat(currResult));
+    }
+
+    return resultArray;
+}
+
+calculateResult = (resultArray) => {
+    let operators = [{'*': (a, b) => a * b,
+                      '/': (a, b) => a / b},
+                     {'+': (a, b) => a + b,
+                      '-': (a, b) => a - b} ];
+    let currResult = [];
+    let currentOp;
+    let indexOfClosingBracket;
+    let openingBracketCounter;
+
+    // Calculating the equations in brackets first
+    for(let i = 0; i < resultArray.length - 1; i++) {
+        indexOfClosingBracket = -1;
+        
+        if(resultArray[i] === '(') {
+            let j = i + 1;
+            openingBracketCounter = 1;
+
+            while(j!== resultArray.length) {
+                if(resultArray[j] === ')') {
+                    openingBracketCounter--;
+                    indexOfClosingBracket = j;
+                    if(openingBracketCounter === 0) {
+                        resultArray.splice(i, indexOfClosingBracket - i + 1, calculateResult(resultArray.slice(i + 1, indexOfClosingBracket)));
```

Comment:
I would maybe split this line into multiple lines, it's just too long to read and understand